trezorlib/tx_api.py

```diff
@@ -65,11 +76,15 @@ def get_tx(self, txhash):
 
 class TxApiInsight(TxApi):
 
-    def __init__(self, network, url=None, zcash=None):
+    def __init__(self, network, url=None, zcash=None, bip115=False):
         super().__init__(network, url)
         self.zcash = zcash
+        self.bip115 = bip115
         if url:
-            self.pushtx_url = url.replace('/api/', '/tx/send')
+            http_protocol, link = url.split('://')
+            if ('/' in link):
+                link, api = link.split('/')
+            self.pushtx_url = http_protocol + '://' + link + '/tx/send'
```

Comment:
no.

What are you trying to accomplish here exactly? Is it that your api server doesn't contain `/api/` in the URL?
I can see that this code is currently broken anyway, because there are no trailing slashes.

Why don't you first strip out trailing `"/api"` if present, and then append `"/tx/send"` always? This splitting on protocol makes no sense.